plugins/devtools-backend/src/api/DevToolsBackendApi.ts

```diff
@@ -0,0 +1,254 @@
+    private readonly config: Config,
+  ) {}
+
+  public async listExternalDependencyDetails(): Promise<ExternalDependency[]> {
+    const result: ExternalDependency[] = [];
+
+    const endpoints = this.config.getOptional<Endpoint[]>(
+      'devTools.externalDependencies.endpoints',
+    );
+    if (!endpoints) {
+      // No external dependency endpoints configured
+      return result;
+    }
+    for (const endpoint of endpoints) {
+      this.logger?.info(
+        `Checking external dependency "${endpoint.name}" at "${endpoint.target}"`,
+      );
+
+      switch (endpoint.type) {
+        case 'ping': {
+          const pingResult = await this.pingExternalDependency(endpoint);
+          result.push(pingResult);
+          break;
+        }
+        case 'fetch': {
+          const fetchResult = await this.fetchExternalDependency(endpoint);
+          result.push(fetchResult);
+          break;
+        }
+        default:
+          return result;
+      }
+    }
+
+    return result;
+  }
+
+  private async fetchExternalDependency(
+    endpoint: Endpoint,
+  ): Promise<ExternalDependency> {
+    let status;
+    let error;
+
+    await fetch(endpoint.target)
+      .then(res => {
+        status =
+          res.status === 200
+            ? ExternalDependencyStatus.healthy
+            : ExternalDependencyStatus.unhealthy;
+        this.logger.debug(
+          `Fetch for ${endpoint.name} resulted in status code "${res.status}"`,
+        );
+      })
+      .catch((err: Error) => {
+        this.logger.error(`Fetch failed for ${endpoint.name} - ${err.message}`);
+        error = err.message;
+      });
+
+    const result: ExternalDependency = {
+      name: endpoint.name,
+      type: endpoint.type,
+      target: endpoint.target,
+      status: status ?? ExternalDependencyStatus.unhealthy,
+      error: error ?? undefined,
+    };
+
+    return result;
+  }
+
+  private async pingExternalDependency(
+    endpoint: Endpoint,
+  ): Promise<ExternalDependency> {
+    const pingResult = await ping.promise.probe(endpoint.target);
+
+    let error;
+    if (
+      pingResult.packetLoss === '100.000' ||
+      pingResult.packetLoss === 'unknown'
+    ) {
+      this.logger.error(
+        `Ping failed for ${endpoint.name} - ${pingResult.output}`,
+      );
+      error =
+        pingResult.output === ''
+          ? `${endpoint.target} - Unknown`
+          : pingResult.output;
+    }
+
+    this.logger.debug(
+      `Ping results for ${endpoint.name}: ${pingResult.output}`,
+    );
+
+    const result: ExternalDependency = {
+      name: endpoint.name,
+      type: endpoint.type,
+      target: endpoint.target,
+      status: pingResult.alive
+        ? ExternalDependencyStatus.healthy
+        : ExternalDependencyStatus.unhealthy,
+      error: error ?? undefined,
+    };
+
+    return result;
+  }
+
+  public async listConfig(): Promise<ConfigInfo> {
+    /* eslint-disable-next-line no-restricted-syntax */
```

Comment:
@rugvip is this the best way to grab all the configs and omit secrets or is there an easier way?

Reply:
There were a couple of comments about this in the initial Draft PR:

- https://github.com/backstage/backstage/pull/14642#discussion_r1025086573
- https://github.com/backstage/backstage/pull/14642#discussion_r1025086987